test/NuGet.Core.Tests/NuGet.Configuration.Test/PackageSourceMappingProviderTests.cs

```diff
@@ -46,6 +46,42 @@ public void GetPackageSourceMappingItems_WithOneConfig_ReturnsCorrectPatterns()
             packageSourceMappingSourceItem.Patterns.First().Pattern.Should().Be("stuff");
         }
 
+        [Fact]
+        public void GetPackageSourceMappingItems_WithOneConfig_WithDuplicateKeys_ReturnsCorrectPatterns()
+        {
+            // Arrange
+            using var mockBaseDirectory = TestDirectory.Create();
+            var configPath1 = Path.Combine(mockBaseDirectory, "NuGet.Config");
+            SettingsTestUtils.CreateConfigurationFile(configPath1, @"<?xml version=""1.0"" encoding=""utf-8""?>
+<configuration>
+    <packageSourceMapping>
+        <packageSource key=""nuget.org"">
```

Comment:
I don't think we'd want to merge the mappings. 

It should 1 section per source and only one should apply.